src/batch-transaction.ts

```diff
@@ -364,12 +379,27 @@ class BatchTransaction extends Snapshot {
    * @example <caption>include:samples/batch.js</caption>
    * region_tag:spanner_batch_execute_partitions
    */
-  execute(partition, callback) {
-    if (is.string(partition.table)) {
-      this.read(partition.table, partition, callback);
+  execute(partition: ReadRequest | ExecuteSqlRequest): Promise<executeResponse>;
```

Comment:
Also modify the documentation as per the response type

Reply:
refactored